### PR TITLE
fix(slack): escape Slack messages to prevent malicious actors from doing things

### PR DESCRIPTION
**Pull request by scefali**
This PR escapes the markdown text that we pass into Slack. It seems the only thing that works is to doubly escape it. Setting `mrkdwn_in` to an empty array didn't work.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -1409,6 +1409,8 @@ def SOCIAL_AUTH_DEFAULT_USERNAME():
     "organizations:org-roles-for-teams": False,
     # Enable new JS SDK Dynamic Loader
     "organizations:js-sdk-dynamic-loader": False,
+    # If true certain Slack messages will be escaped to prevent rendering markdown
+    "organizations:slack-escape-messages": False,
```
Comment (evanpurkhiser) [inline src/sentry/conf/server.py:1413]: Why does this need to be a feature flag

Comment (scefali) [reply src/sentry/conf/server.py:1413]: @evanpurkhiser slightly worried it's going to break formatting and I want to test internally first

Comment (evanpurkhiser) [reply src/sentry/conf/server.py:1413]: cool, let's just make sure it get's cleaned up 😆

Comment (scefali) [reply src/sentry/conf/server.py:1413]: @evanpurkhiser lol yea well i def have to remove this once it's ready to go GA

File: src/sentry/integrations/slack/utils/escape.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
+from __future__ import annotations
+
 # TODO(scttcper): Might need to handle "*" bold, and "_" italics
 translator = str.maketrans({"&": "&amp;", "<": "&lt;", ">": "&gt;"})
 
 
-def escape_slack_text(txt: str) -> str:
+def escape_slack_text(txt: str | None) -> str:
+    if not txt:
+        return ""
```
Comment (markstory) [inline src/sentry/integrations/slack/utils/escape.py:9]: Should this be under the functions docstring?

**Review by dashed — APPROVED** [review]

File: src/sentry/integrations/slack/utils/escape.py (hunk 2 above)
Comment (scefali) [reply src/sentry/integrations/slack/utils/escape.py]: fixed

**Pull request merged**